lib/plugins/bed.js

```diff
@@ -24,8 +24,18 @@ function inject (bot) {
     } else if (bedBlock.type !== 26) {
       cb(new Error('wrong block : not a bed block'))
     } else {
-      bot.activateBlock(bedBlock)
-      cb()
+      bot.activateBlock(bedBlock, () => {
+        const sleepTimeout = setTimeout(() => {
+          const sleepCb = bot.listeners('sleep').find(f => f.name === 'sleepCb')
+          bot.removeListener('sleep', sleepCb)
+          cb(new Error('there are monsters nearby'))
+        }, 500)
```

Comment:
I don't think that a timeout would do it. What if bot has high ping and exceed the timeout?

Reply:
Would 1.5s be okay or what value would you suggest?

Reply:
So what about [this](https://github.com/PrismarineJS/mineflayer/issues/568#issuecomment-578296191) solution?